### PR TITLE
Integration with AI/ML API

### PR DESCRIPTION
**Pull request by waterstark**
Hi!
I am a Python developer working at AI/ML API.
This PR is needed to add the ability to send requests to our models.

### CODE REVIEW

**Comment by cjbbb** [comment]
There seems to be a conflict, and the link "https://api.aimlapi.com/v2" cannot be opened. 
By the way, may I ask what is the advantages of "AI/ML" api

**Comment by waterstark** [comment]
> There seems to be a conflict, and the link "https://api.aimlapi.com/v2" cannot be opened. By the way, may I ask what is the advantages of "AI/ML" api

This link requires a POST request and needs to be slightly modified if you want to use it outside your app. For example: POST https://api.aimlapi.com/v2/chat/completions.

Our main advantage is that we provide access to over 200 models, covering a wide range of closed- and open-source options. This includes models like Qwen, LLaMA, OpenAI, Anthropic, Grok, and Mistral. With us, users can access their favorite LLMs easily, with minimal rate limits (even for o1 models) and no regional restrictions.

If you'd like more information, feel free to visit our website:
https://aimlapi.com/

**Comment by cjbbb** [comment]
@waterstark  Thanks! Could you solve the conflicts so I can approve?

**Comment by surapuramakhil** [comment]
@waterstark can you update ReadMe about this https://github.com/AIHawk-app/Auto_Jobs_Applier/pull/757#issuecomment-2459753726

**Review by surapuramakhil — CHANGES_REQUESTED** [review]
Based on these comments can you exact url which work outside of their app in ReadMe.

https://github.com/AIHawk-co/Auto_Jobs_Applier/pull/757#issuecomment-2459753726
https://api.aimlapi.com/v2/chat/completions.

**Review by surapuramakhil — DISMISSED** [review]

**Comment by surapuramakhil** [comment]
@waterstark this PR will be closed if conflicts are resolved.
PR will close as Stale.

**Comment by waterstark** [comment]
We want to show users which endpoint needs to be use within their local JobsApplier app - https://api.aimlapi.com/v2.
It is the one that we propose should be added to readme (alr added in the current pr)

If you are wondering why it cannot be opened - this link does not need to be opened by itself, it is an endpoint that has to receive a proper body+headers, in which case it responds with results of a generation, giving users access to the AI models

**Comment by surapuramakhil** [comment]
@waterstark here it goes

- [ ] there are conflicts which are blocking merge
- [ ] is endpoint in llm_manager correct?

File: src/llm/llm_manager.py [hunk 1]
```diff
@@ -33,20 +32,40 @@ def invoke(self, prompt: str) -> str:
...
         logger.debug("Invoking OpenAI API")
         response = self.model.invoke(prompt)
         return response
 
 
+class AIMLModel(AIModel):
+    def __init__(self, api_key: str, llm_model: str):
+        from langchain_openai import ChatOpenAI
+
+        self.base_url = "https://api.aimlapi.com/v2"
```
Comment (surapuramakhil) [inline src/llm/llm_manager.py:50]: this endpoint

Comment (waterstark) [reply src/llm/llm_manager.py:50]: Yes, the URL is correct, and I tested the basic functionality of the app—it was working.

**Review by waterstark — COMMENTED** [review]
Base support for AI/ML API

**Comment by surapuramakhil** [comment]
@waterstark let us know once conflicts are resolved. we are happy to merge.

**Review by surapuramakhil — APPROVED** [review]

**Comment by OctavianTheI** [comment]
Hi guys!
I'm Sergey from aimlapi, working with Ivan on the integrations.
It seems that this pr was lost, and hasn't reached the main branch after all :)
Could you include these in a future release?

**Comment by waterstark** [comment]
Hi Akhil and CJ!

Ivan from AI/ML API here
It seems that the branch feder-cr:release/v3.1.0 was not merged with main

Can you give me a link to the upcoming release branch, so that i could commit the changes there?

**Comment by OctavianTheI** [comment]
Hi, @feder-cr!
can you link us the branch, where we could transfer these changes?
Our integration, and the integration from Groq never reached main - so we'd love to transfer our work to a more relevant part of the repo

**Pull request merged**